skimage/transform/_thin_plate_splines.py

```diff
@@ -0,0 +1,325 @@
+    --------
+    >>> import skimage as ski
+
+    Define source and destination points and generate meshgrid for transformation:
+
+    >>> src = np.array([[0, 0], [0, 5], [5, 5],[5, 0]])
+    >>> dst = np.roll(src, 1, axis=0)
+    >>> coords = np.meshgrid(np.arange(5), np.arange(5))
+    >>> t_coords = np.vstack([coords[0].ravel(), coords[1].ravel()]).T
+
+    >>> tps = ski.transform.TpsTransform()
+    >>> tps.estimate(src, dst)
+    True
+
+    Apply the transformation
+
+    >>> trans_coord = tps(t_coords)
+    >>> xx_trans = trans_coord[:, 0]
+    >>> yy_trans = trans_coord[:, 1]
+    >>> coords[1]
+    array([[0, 0, 0, 0, 0],
+           [1, 1, 1, 1, 1],
+           [2, 2, 2, 2, 2],
+           [3, 3, 3, 3, 3],
+           [4, 4, 4, 4, 4]])
+    >>> expected_yy = np.array([0, 1, 2, 3, 4,
+    ...                         0, 1, 2, 3, 4,
+    ...                         0, 1, 2, 3, 4,
+    ...                         0, 1, 2, 3, 4,
+    ...                         0, 1, 2, 3, 4])
+    >>> np.allclose(yy_trans, expected_yy)
+    True
+    """
+    def __init__(self):
+        self._estimated = False
+        self.parameters = None
+        self.src = None
+
+    def __call__(self, coords):
+        """Estimate the transformation from a set of corresponding points.
+
+        Parameters
+        ----------
+        coords : (N, 2) array_like
+            x, y coordinates to transform
+
+        Returns
+        -------
+        transformed_coords: (N, D) array
+            Destination coordinates
+        """
+        if self.parameters is None:
+            raise ValueError(f"{self.parameters}. Compute the `estimate`")
+        coeffs = self.parameters
+        coords = np.array(coords)
+
+        if not coords.ndim == 2 or coords.shape[1] != 2:
+            raise ValueError("Input 'coords' must have shape (N,2)")
+
+        x_warp = self._spline_function(coords[:, 0], coords[:, 1], coeffs[:, 0])
+        y_warp = self._spline_function(coords[:, 0], coords[:, 1], coeffs[:, 1])
+        return np.vstack([x_warp, y_warp]).T
+
+    @property
+    def inverse(self):
+        raise NotImplementedError("This is yet to be implemented.")
+
+    def estimate(self, src, dst):
+        """Estimate how close is the deformed source to the target.
+
+        Number of source and destination points must match.
+
+        Parameters
+        ----------
+        src : (N, 2) array_like
+            Control point at source coordinates
+        dst : (N, 2) array_like
+            Control point at destination coordinates
+
+        Returns
+        -------
+        success: bool
+            True, if all pieces of the model are successfully estimated.
+        """
+
+        src = _ensure_2d(src)
+        dst = _ensure_2d(dst)
+
+        if src.shape != dst.shape:
+            raise ValueError("src and dst shape must be identical")
+
+        # if src.shape[-1] != 2 and dst.shape[-1] != 2:
+        #     raise ValueError("src and dst must have shape (N,2)")
```

Comment:
Is this commented out on purpose?

Reply:
Yes, but it's no longer relevant for this implementation